File: players_writeup/1047/1to-submit/algo-complexity/dinic-gen.py
import random
random.seed(0)

n = 100
edges = []
MX = 10000
for i in range(1, n-1):
	edges.append((i, i+1, MX))
	edges.append((i+1, n, 1))
edges.append((n-1, n, MX))

while len(edges) < 5000:
	a = random.randint(1, n)
	b = random.randint(1, n)
	if a == b:
		continue
	edges.append((a, b, random.randint(1, 2)))

m = len(edges)
print(n, m, 1, n)
for e in edges:
	print(*e)
	

File: players_writeup/1244/algo-spfa/dinic_gen.cpp
#include <iostream>
#include <tuple>
#include <vector>
#include <random>
#include <algorithm>

const int N = 50;
// const int MAXW = 124000;
// const int SEED = 40;

using Edge = std::tuple<int, int, int>;

int main(void) {
    std::vector<Edge> edges;
    int n = N * 2;
    for (int i = 1; i <= N; i++) {
        for (int j = N + 1; j <= N * 2; j++) {
            edges.push_back({i, j, 1});
            if (i < N)
                edges.push_back({i, j, 1});
        }
        edges.push_back({i, i + 1, n});
    }
    std::cout << n << " " << edges.size() << " " << 1 << " " << n << std::endl;
    for (auto e : edges) {
        std::cout << std::get<0>(e) << " " << std::get<1>(e) << " " << std::get<2>(e) << std::endl;
    }
    return 0;
}
